pkg/controller/networkpolicy/controller_query_test.go

```diff
@@ -0,0 +1,191 @@
+	_, controller := newController(objects...)
+	// create querier with stores inside controller
+	address, appliedToGroup, policy, podInformer := controller.addressGroupStore, controller.appliedToGroupStore, controller.internalNetworkPolicyStore, controller.podInformer
+	querier := NewEndpointQueryReplier(address, appliedToGroup, policy, podInformer)
+
+	return controller, querier
+}
+
+// TestInvalidSelector tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) does not
+// select any pods
+func TestInvalidSelector(t *testing.T) {
+	_, endpointQuerier := makeControllerAndEndpointQueryReplier()
+	// test appropriate response to QueryNetworkPolices
+	namespace, pod := "non-existing-namespace", "non-existing-pod"
+	response := endpointQuerier.QueryNetworkPolicies(namespace, pod)
+
+	assert.Equal(t, errors.NewNotFound(v1.Resource("pod"), pod), response.Error, "expected not found error")
+}
+
+// TestSingleAppliedPolicy tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) selects a
+// pod which has a single networkpolicy object applied to it
+func TestSingleAppliedPolicy(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	controller, endpointQuerier := makeControllerAndEndpointQueryReplier(&namespaces[0], &pods[0], &policies[0])
+
+	controller.informerFactory.Start(ctx.Done())
+	controller.crdInformerFactory.Start(ctx.Done())
+
+	go controller.NetworkPolicyController.Run(ctx.Done())
+
+	time.Sleep(2 * time.Second)
+
+	namespace1, pod1 := "testNamespace", "podA"
+	response1 := endpointQuerier.QueryNetworkPolicies(namespace1, pod1)
+	require.Equal(t, nil, response1.Error)
+
+	assert.Equal(t, response1.Endpoints[0].Policies[0].PolicyRef.Name, "default-deny-ingress")
+}
+
+// TestSingleEgressPolicy tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) selects
+// a pod which has a single networkpolicy which defines an egress policy from it.
+func TestSingleEgressPolicy(t *testing.T) {
+	assert.Fail(t, "unimplemented")
+}
+
+// TestSingleIngressPolicy tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) selects
+// a pod which has a single networkpolicy which defines an ingress policy from it.
+func TestSingleIngressPolicy(t *testing.T) {
+	assert.Fail(t, "unimplemented")
+}
+
+// TestMultiplePolicy tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) selects
+// a pod which has multiple networkpolicies which define policies on it.
+func TestMultiplePolicy(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	controller, endpointQuerier := makeControllerAndEndpointQueryReplier(&namespaces[0], &pods[0], &policies[0], &policies[1])
+
+	controller.informerFactory.Start(ctx.Done())
+	controller.crdInformerFactory.Start(ctx.Done())
+
+	go controller.NetworkPolicyController.Run(ctx.Done())
+
+	time.Sleep(2 * time.Second)
```

Comment:
hopefully we can find a way to unify this code instead of duplicating it in all test cases